fastlane/lib/fastlane/actions/apple_pay_cert.rb

```diff
@@ -0,0 +1,153 @@
+require 'spaceship'
+
+module Fastlane
+  module Actions
+    module SharedValues
+    end
+
+    class ApplePayCertAction < Action
+      def self.run(params)
+        UI.message("Starting login with user '#{params[:username]}'")
+        Spaceship.login(params[:username], nil)
+        Spaceship.client.select_team(team_id: params[:team_id], team_name: params[:team_name])
+        UI.message("Successfully logged in")
+
+        create_certificate(params)
+      end
+
+      def self.create_certificate(params)
+        UI.important("Creating a new Apple Pay certificate.")
+        csr, pkey = Spaceship.certificate.create_apple_pay_certificate_signing_request
+
+        begin
+          cert = certificate.create!(csr: csr, bundle_id: params[:merchant_bundle_id])
+        rescue => ex
+          if ex.to_s.include?("You already have a current")
+            UI.message(ex.to_s)
+            UI.user_error!("You already have 2 apple pay certificates for this merchant. You'll need to revoke an old certificate to make room for a new one")
+          else
+            raise ex
+          end
+        end
+
+        x509_certificate = cert.download
+        filename_base = params[:merchant_bundle_id] # PEM.config[:pem_name] || "#{PEM.config[:app_identifier]}"
+        filename_base = File.basename(filename_base, ".pem") # strip off the .pem if it was provided.
+
+        output_path = File.expand_path('.')
+        FileUtils.mkdir_p(output_path)
+
+        if params[:save_private_key]
+          private_key_path = File.join(output_path, "#{filename_base}.pkey")
+          File.write(private_key_path, pkey.to_pem)
+          UI.message("Private key: ".green + Pathname.new(private_key_path).realpath.to_s)
+        end
+
+        if params[:generate_p12]
+          p12_cert_path = File.join(output_path, "#{filename_base}.p12")
+          p12 = OpenSSL::PKCS12.create(params[:p12_password], 'production', pkey, x509_certificate)
+          File.write(p12_cert_path, p12.to_der)
+          UI.message("p12 certificate: ".green + Pathname.new(p12_cert_path).realpath.to_s)
+        end
+
+        x509_cert_path = File.join(output_path, "#{filename_base}.pem")
+        File.write(x509_cert_path, x509_certificate.to_pem + pkey.to_pem)
+        UI.message("PEM: ".green + Pathname.new(x509_cert_path).realpath.to_s)
+        return x509_cert_path
+      end
+
+      def self.certificate
+        Spaceship.certificate.apple_pay_certificate
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.is_supported?(platform)
+        [:ios].include?(platform)
+      end
+
+      def self.description
+        'Creates Apple Pay certificate'
+      end
+
+      def self.details
+        "This action allows you to create Apple Pay merchant certificate."
```

Comment:
What is the difference between an "Apple Pay certificate" and an "Apple Pay merchant certificate"? Are these identical? Are there any other "Apple Pay vertificates" this could clash with?

Reply:
Does this actually refer to this process?
> Create a payment processing certificate
https://help.apple.com/developer-account/#/devb2e62b839?sub=devf31990e3f